qlib/Swagger.qm

```diff
@@ -0,0 +1,3666 @@
+            requirements). Individual operations can override this definition.
+
+            Individual values are hashes in the form of Security Requirement Objects.
+            See https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#securityRequirementObject.
+            That is, the hash keys MUST correspond to security schemes declared
+            in the @ref SwaggerSchema::securityDefinitions "Security Definitions",
+            while the values are lists of scope names (strings) required for the execution.
+         */
+        hash<string, softlist<string>> security;
+
+        #! A list of @ref TagObject "tags" used by the specification with additional metadata.
+        /**
+            The order of the tags can be used to reflect on their order by the
+            parsing tools. Not all tags that are used by the @ref OperationObject
+            must be declared. The tags that are not declared may be organized
+            randomly or based on the tools' logic. Each tag name in the list
+            MUST be unique.
+
+            Individual values in this list are of type @ref TagObject "TagObject".
+         */
+        list<TagObject> tags;
+
+        #! Additional external documentation.
+        *ExternalDocumentationObject externalDocs;
+
+        #! SwaggerSchema options
+        const SwaggerOptions = (
+            "compact_serialization": True,
+            "def_path": True,
+            "try_import": True,
+        );
+    }
+
+    private {
+        #! if serialized data should be subject to compact serialization (default: @ref Qore::True "True")
+        bool compact_serialization;
+
+        #! the default path to use when retrieving external schema references
+        *string def_path;
+
+        #! a call reference or closure to be passed a string name for external schema references, must take a string argument (the resource name) and return a string (the resource data)
+        *code try_import;
+    }
+
+    #! Builds the schema representation from the deserialized schema hash describing the root document object
+    /**
+        @param oh deserialized hash from the source schema description describing the root document object
+        @param opts options as per @ref SwaggerOptions as follows:
+        - \c compact_serialization (default @ref Qore::True "True"): optimize the size of serialized data strings
+        - \c def_path: the default path to use when retrieving external schema references
+        - \c try_import: a call reference or closure to be passed a string name for external schema references, must take a string argument (the resource name) and return a string (the resource data)
+
+        @throws INVALID-FIELD-FORMAT field has invalid format
+        @throws INVALID-FIELD-TYPE field has invalid type
+        @throws REQUIRED-FIELD-MISSING required field is missing
+        @throws SECURITY-ERROR inconsistent security information in schema
+     */
```

Comment:
`SWAGGER-OPTION-ERROR`, `INVALID-SWAGGER-VERSION` and `INVALID-FIELD-VALUE` exceptions are missing here.

Reply:
done